sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,372 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import logging
+
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+if daal_check_version((2024, "P", 600)):
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.linear_model import Ridge as sklearn_Ridge
+    from sklearn.metrics import r2_score
+    from sklearn.utils.validation import check_is_fitted, check_X_y
+
+    from daal4py.sklearn.linear_model._ridge import _fit_ridge as daal4py_fit_ridge
+
+    if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+        from sklearn.linear_model._base import _deprecate_normalize
+
+    from onedal.linear_model import Ridge as onedal_Ridge
+    from onedal.utils import _num_features, _num_samples
+
+    from ..._device_offload import dispatch, wrap_output_data
+    from ..._utils import PatchingConditionsChain
+
+    def _is_numeric_scalar(value):
```

Comment:
nvm I see that you have added it - you should be able to remove definition and use of _is_numeric_scalar() now

Reply:
used to validate that alpha is numeric since it can also be array-like for multitarget regression but our oneDAL implementation lacks this functionality yet. please resolve if this makes sense